dm/syncer/validator_cond.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syncer
+
+import (
+	"strings"
+)
+
+type Cond struct {
+	Table     *validateTableInfo
+	ColumnCnt int
+	PkValues  [][]string
```

Comment:
can we reuse RowChange instead ? see https://github.com/pingcap/tiflow/blob/master/pkg/sqlmodel/row_change.go

Reply:
struct is similar, but nearly no code can be reused, also there're fields special to validation, will leave it before all logic is completed.

Reply:
the main inconvenience of RowChange.genWhere is it can't handle INSERT type binlog event. I think we can extract a more generic genWhere and let RowChange and validator use it.

There is also a similar logic of CDC in `prepareDelete`, I really hope we can unify them.